src/App.js

```diff
@@ -1,75 +1,159 @@
+import Header from './Components/Header/Header';
+import Todoslist from './Components/Todoslist/Todoslist';
+import Footer from './Components/Footer/Footer';
 
-function App() {
-  return (
-    <section className="todoapp">
-      <header className="header">
-        <h1>todos</h1>
+let id = 0;
 
-        <input
-          className="new-todo"
-          placeholder="What needs to be done?"
-        />
-      </header>
-
-      <section className="main" style={{ display: 'block' }}>
-        <input type="checkbox" id="toggle-all" className="toggle-all" />
-        <label htmlFor="toggle-all">Mark all as complete</label>
-
-        <ul className="todo-list">
-          <li className="">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-1" />
-              <label htmlFor="todo-1">sdfsdfsdf</label>
-              <button type="button" className="destroy" />
-            </div>
-          </li>
-
-          <li className="">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-2" />
-              <label htmlFor="todo-2">sakgjdfgkhjasgdhjfhs</label>
-              <button type="button" className="destroy" />
-            </div>
-          </li>
-
-          <li className="">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-3" />
-              <label htmlFor="todo-3">sddfgdfgdf</label>
-              <button type="button" className="destroy" />
-            </div>
-          </li>
-        </ul>
-      </section>
+class App extends React.Component {
+  state = {
+    todo: '',
+    todosList: [],
+    filteredTodosList: [],
+    filterState: '',
+  };
+
+  handleChange = ({ target }) => {
+    const { value } = target;
+
+    this.setState({
+      todo: value,
+    });
+  }
+
+  handleSubmit = (todo) => {
+    if (todo) {
+      const todoItem = {
+        todoTitle: todo,
+        id,
+        completed: false,
+      };
+
+      id += 1;
+
+      this.setState(prevState => ({
+        ...prevState,
+        todo: '',
+        todosList: [...prevState.todosList, todoItem],
+        filteredTodosList: [...prevState.todosList, todoItem],
+      }));
+    }
+  };
+
+  handleCheck = ({ target }, todoItem) => {
+    this.setState(prevState => ({
+      todosList: prevState.todosList.map((todo) => {
+        if (todo.id === todoItem.id) {
+          return { ...todo, completed: target.checked };
+        }
+
+        return todo;
+      }),
+      filteredTodosList: prevState.filteredTodosList.map((todo) => {
+        if (todo.id === todoItem.id) {
+          return { ...todo, completed: target.checked };
+        }
+
+        return todo;
+      }),
+    }));
+  }
+
+  handleDestroy = (todoItem) => {
+    this.setState(prevState => ({
+      todosList: prevState.todosList.filter((todo) => {
+        if (todo.id === todoItem.id) {
+          return false;
+        }
+
+        return todo;
```

Comment:
в условии фильтера достаточно вернуть `true` или `false` 
```suggestion
        return true;
```